### DISCUSSION TITLE
Support for Metal M1

### DISCUSSION THREAD

**Discussion by awesomegogos** [gomlx/gomlx#79] [Q&A]
Hi, it was mentioned in the latest release that there's support for metal gpu? Would you guys have a sample code to do that? I'm getting this error 

<img width="1178" alt="image" src="https://github.com/user-attachments/assets/ec43d3dc-0dec-42e8-9d9e-bcbf4d34d9dc">

**Comment by janpfeifer** [comment]
hi @mabreyes, thanks for reporting this.

I ended up naming the Apple/Metal PJRT also "cpu" ([see here in the `install_darwin.sh` script](https://github.com/gomlx/gopjrt/blob/main/cmd/install_darwin.sh#L50)) because it's the default one, even though I think it is a mix of CPU/Metal PJRT -- the PJRT plugin code is closed-sourced, so I'm not 100% sure.

Could you try setting `GOMLX_BACKEND="xla:cpu"` instead, or just not setting it ? 

Just a reminder: the Apple/Metal support is very alpha (experimental). There are many things that Apple's PJRT plugin (distributed for Jax) doesn't support. So many tests don't work -- many simply because it doesn't support Float64 -- but I haven't had yet the time to debug much. There may be other issues that I'm not aware of. I managed to run many tests just fine though. 

Sadly, I don't have an Apple hardware easily available -- I have to rent a cloud Mac Mini every-time, so Mac development goes slower.

Let me know what you figure out.
